status/versions.go

```diff
@@ -0,0 +1,116 @@
+package status
+
+import (
+	"encoding/json"
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"io/ioutil"
+	"k8s.io/api/core/v1"
+	kube "k8s.io/client-go/kubernetes"
+	"net/http"
+	"regexp"
+)
+
+type serviceProduct func() (*ProductInfo, error)
+
+func getVersions() {
+	products := []serviceProduct{
+		istioVersion,
+		prometheusVersion,
+		kubernetesVersion,
+	}
+	for _, prod := range products {
+		getVersionProduct(prod)
+	}
+}
+
+func getVersionProduct(serviceProduct serviceProduct) {
+	productInfo, err := serviceProduct()
+	if err == nil {
+		info.Products = append(info.Products, *productInfo)
+	}
+}
+
+func istioVersion() (*ProductInfo, error) {
+	product := ProductInfo{}
+	istioConfig := config.Get().Products.Istio
+	serviceInfo, err := getService(istioConfig.ServiceNamespace, istioConfig.ServiceVersion)
+	if err == nil {
+		resp, err := http.Get("http://" + serviceInfo.ClusterIP + ":9093/version")
```

Comment:
That http call doesn't sound reliable, what if the port is not 9093? Are we sure that ClusterIP will always be reachable like this?
If there is no easy solution to apply right now we could keep that as a best effort and create a JIRA issue to mention it, ok?

Reply:
Mmm by default the port is 9093, I don't know if someone will install Istio in another port in a container, we can add a port configuration in the config Istio.

Reply:
I would also add the URL http://host:version/ as a whole parameter.
I think we are now using istio without TLS, but once we jump into security, perhaps those services will be exposed under a https protocol.
So, better to config them easily instead from code.

Reply:
Also, this is a side comment, not related to this PR.
As we need to differentiate between istio.io builts and redhat builts we need to check if this service is enough or probably we will need an additional check to mark 0.7.1 from redhat image or whatever.
I haven't touched the RH bits but as soon as QE can check, we can confirm it.

(Nothing requested about this for this particular PR).

Reply:
I'm not really sure about this, but, can't we use the Istio proxy to get this? 

https://github.com/istio/istio/blob/master/pilot/doc/proxy-controller.md#discovery-service

I recall i read somewhere that we could communicate to other nodes in the mesh trough the proxy (envoy) without really knowing their ip address, but again, not sure about this.

Reply:
Just about the port, they may not be the container port, but the service port (not necessarily the same). It can be taken from there : `Service.Spec.Ports` from kubernetes service model

Reply:
I suggest that if we cannot find the version, be it because we cannot reach the URL, parse the version or whatever else we just fall back to "unknown" version. Actually to do so you could just initialize versions with "unknown".

Reply:
I did the changes @lucasponce @jotak Thanks.